packages/dds/tree/src/events/events.ts

```diff
@@ -67,44 +64,46 @@ export type Events<E> = {
  * }
  * ```
  */
-export type TransformEvents<E extends Events<E>, Target extends IEvent = IEvent> = {
-	[P in keyof Events<E>]: (event: P, listener: E[P]) => void;
+export type TransformListeners<
+	TListeners extends Listeners<TListeners>,
+	Target extends IEvent = IEvent,
```

Comment:
Nit: since the other type parameter starts with a non-name "T", "Target" probably wants to become "TTarget"?

Reply:
Sounds good to me.